### PR TITLE
Added recipe for pytest-astropy-header

### PR DESCRIPTION
**Pull request by astrofrog**
Checklist

* [x] License file is packaged (see here for an example)
* [x] Source is from official source
* [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
* [ ] <s>If static libraries are linked in, the license of the static library is packaged.</s>
* [x] Build number is 0
* [x] A tarball (url) rather than a repo (e.g. git_url) is used in your recipe (see here for more details)
* [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by astrofrog-conda-forge** [comment]
I'm happy to be listed as a maintainer

File: recipes/pytest-astropy-header/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  license: BSD 3-Clause
+  license_file: LICENSE.rst
+  summary: 'Pytest plugin to add diagnostic information to the header of the test output'
+  doc_url: {{ git_url }}
+  dev_url: {{ git_url }}
+
+extra:
+  recipe-maintainers:
+    - astrofrog-conda-forge
+    - bsipocz
```
Comment (scopatz) [inline recipes/pytest-astropy-header/meta.yaml]: @bsipocz Are you OK with being a maintainer?

Comment (bsipocz) [reply recipes/pytest-astropy-header/meta.yaml]: yes, I'm, in any feedstocks coming from astropy/astrofrog/etc.

**Comment by astrofrog** [comment]
This should be good to go!

File: recipes/pytest-astropy-header/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
```
Comment (marcelotrevisani) [inline recipes/pytest-astropy-header/meta.yaml]:
```suggestion
  script: "{{ PYTHON }} -m pip install . -vv"
```

**Pull request merged**